internal/accounts/sql.go

```diff
@@ -13,29 +13,143 @@ const (
 	schemaName      = `kwild_accts`
 	sqlCreateSchema = `CREATE SCHEMA IF NOT EXISTS ` + schemaName
 
+	accountStoreVersion = 1
+
 	sqlInitTables = `CREATE TABLE IF NOT EXISTS ` + schemaName + `.accounts (
 		identifier BYTEA PRIMARY KEY,
 		balance TEXT NOT NULL, -- consider: NUMERIC(32) for uint256 and pgx.Numeric will handle it and provide a *big.Int field
 		nonce BIGINT NOT NULL -- a.k.a. INT8
 	);`
 
+	// accountsVersionTable is a table that stores the version of the account store.
+	// This is a single row table with a name and version column. The name is always 'version'.
+	accountsVersionTable = `CREATE TABLE IF NOT EXISTS ` + schemaName + `.version (
+		name TEXT NOT NULL PRIMARY KEY, -- name: 'version'
+		version INT NOT NULL
+	);`
+
 	sqlCreateAccount = `INSERT INTO ` + schemaName + `.accounts (identifier, balance, nonce) VALUES ($1, $2, $3)`
 
 	sqlUpdateAccount = `UPDATE ` + schemaName + `.accounts SET balance = $1, nonce = $2
 		WHERE identifier = $3`
 
 	sqlGetAccount = `SELECT balance, nonce FROM ` + schemaName + `.accounts WHERE identifier = $1`
+
+	upsertVersion = `INSERT INTO ` + schemaName + `.version (name, version) VALUES ('version', $1) ON CONFLICT(name) DO UPDATE SET version = $1;`
+
+	getVersion = `SELECT version FROM ` + schemaName + `.version WHERE name = 'version';`
+
+	acctsExists = `SELECT EXISTS (SELECT 1 FROM information_schema.tables WHERE table_schema = ` + schemaName + ` AND table_name = 'accounts');`
 )
 
-func initTables(ctx context.Context, tx sql.DB) error {
+func initOrUpgradeTables(ctx context.Context, db sql.DB) error {
+	// Get the current version of the account store.
+	curVersion, versionErr := getVersionNumber(ctx, db)
+	if versionErr != nil {
```

Comment:
Based on the implementation of `getVersionNumber`, it seems like we would want to fail in case of an error here? It would suggest that there is something critically wrong